src/main/scala/chisel3/simulator/package.scala

```diff
@@ -175,12 +175,14 @@ package object simulator {
               }
             }
             case element: Element =>
-              DataMirror.directionOf(element) match {
+              val _p = DataMirror.directionOf(element) match {
                 case ActualDirection.Input =>
                   Seq((element, ModuleInfo.Port(name, isGettable = true, isSettable = true)))
                 case ActualDirection.Output => Seq((element, ModuleInfo.Port(name, isGettable = true)))
                 case _                      => Seq()
               }
+              // Return the port only if the width is positive (firtool will optimized it out from the *.sv primary source)
+              if (element.widthKnown && element.getWidth > 0) _p else Seq()
```

Comment:
```suggestion
              // Return the port only if the width is positive (firtool will optimized it out from the *.sv primary source)
              if (element.widthKnown && element.getWidth > 0) {
                DataMirror.directionOf(element) match {
                  case ActualDirection.Input =>
                    Seq((element, ModuleInfo.Port(name, isGettable = true, isSettable = true)))
                  case ActualDirection.Output => Seq((element, ModuleInfo.Port(name, isGettable = true)))
                  case _                      => Seq()
                }
              } else {
                Seq()
              }
```
Better to not bother running the logic to generate the Port info if we don't need to.